### PR TITLE
[Package Management] Enable internal NuGet publishing pipeline

### PR DESCRIPTION
**Pull request by markavitale**
Add an ADO build definition to publish a NuGet package to an internal Microsoft feed.

General outline of the process
1. Use Xcode 11.2.1 as the officially supported Xcode version for this repo
1. Build each platform, specifically the static libs for iOS
1. Zip up all the relevant build output
1. Optionally add a branch specifier to the build version if we're not on master to indicate a non-standard release
1. Pack and Push our nupkg

Delete plenty of unused files from the previous platform-specific NuGet publish tasks

Note: This is an internal implementation detail of Microsoft's consumption of Fluent UI Apple and is not used for external clients. Instead, please use Carthage or CocoaPods to consume this repository.

For internal reviewers: The build pipeline can be found [here](https://dev.azure.com/ms/ui-fabric-react-native/) and the NuGet package can be found [here](https://dev.azure.com/Office/ISS/_packaging?_a=package&feed=Office&view=versions&package=Microsoft.FluentUI.Apple&version=1.2020.418-v9-nugetPublishing&protocolType=NuGet)

### CODE REVIEW

File: .ado/templates/apple-xcode-build.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+steps:
+  - task: Xcode@5
+    displayName: 'Xcode ${{ parameters.xcode_actions }} ${{ parameters.xcode_configuration }} ${{ parameters.xcode_sdk }} ${{ parameters.xcode_scheme }}'
+    inputs:
+      actions: '${{ parameters.xcode_actions }}'
+      configuration: ${{ parameters.xcode_configuration }}
+      sdk: ${{ parameters.xcode_sdk }}
+      xcWorkspacePath: ${{ parameters.xcode_workspacePath }}
+      scheme: ${{ parameters.xcode_scheme }}
+      xcodeDeveloperDir: /Applications/Xcode_11.2.1.app/Contents/Developer
```
Comment (anandrajeswaran) [inline .ado/templates/apple-xcode-build.yml:19]: Is there a way to extract the xcode version? Not sure if this is the only place we specify this, but it doesn't seem intuitive for someone to go easily update when we get to 11.4

Comment (anandrajeswaran) [reply .ado/templates/apple-xcode-build.yml:19]: Update - it's not the only place. Just saw the apple-xcode-select script

File: scripts/prepare_for_nuget_pack.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+# Copy each platform. Note that we're omitting binary swiftmodule files in favor of swiftinterface files.
+make_dir_if_necessary "nuget/Debug-macosx"
+rsync -a --exclude '*.swiftsourceinfo' --exclude '*.swiftdoc' --exclude '*.swiftmodule' Debug/FluentUI.framework/ nuget/Debug-macosx/FluentUI.framework/
+
+make_dir_if_necessary "nuget/Ship-macosx"
+rsync -a --exclude '*.swiftsourceinfo' --exclude '*.swiftdoc' --exclude '*.swiftmodule' Release/FluentUI.framework/ nuget/Ship-macosx/FluentUI.framework/
+
+make_dir_if_necessary "nuget/Debug-iphoneos"
+rsync -a Debug-iphoneos/libFluentUILib.a nuget/Debug-iphoneos/
+rsync -a --exclude 'Project/' --exclude '*.swiftdoc' --exclude '*.swiftmodule' Debug-iphoneos/FluentUILib.swiftmodule/ nuget/Debug-iphoneos/FluentUILib.swiftmodule/
```
Comment (anandrajeswaran) [inline scripts/prepare_for_nuget_pack.sh:44]: the --excludes seem heavily repeated. Is there a way to factor out?

File: .ado/templates/apple-xcode-build.yml (hunk 1 above)
Comment (markavitale) [reply .ado/templates/apple-xcode-build.yml]: This logic is now centralized in scripts/xcode_select_current_version.sh for both GitHub Actions CI and the new ADO NuGet publishing pipeline.

File: scripts/prepare_for_nuget_pack.sh (hunk 2 above)
Comment (markavitale) [reply scripts/prepare_for_nuget_pack.sh]: This has been extracted into a bash variable.

**Comment by markavitale** [comment]
Noticed that our use of `--exclude '*.swiftmodule'` in our framework copy phase loses our swiftinterface files as they're included in a FluentUI.swiftmodule folder. Looking into a fix.

**Pull request merged**